### ISSUE TITLE
Proxy request errors with venia-pwa-concept

### ISSUE THREAD

**Issue by ericerway** [magento-research/venia-pwa-concept#56]
Issue reported from Mike Stacey in #pwa on 5/3 at 8:41pm CT

---

I'm trying to get the pwa theme up and running on a dev site. My site is running on `http://magento222.test/` and when i run the `npm start` the process gets going, but i notice that it prints this:

`Project is running at https://magento-venia.local.pwadev:8000/
webpack output is served from https://magento-venia.local.pwadev:8000/pub/static/frontend/Magento/venia/en_US`

When i get the url `https://magento-venia.local.pwadev:8000/` the app does run, but i get a host of errors:

Error: ENOENT: no such file or directory, stat '/Applications/MAMP/htdocs/venia-pwa-concept/theme-frontend-venia/web/js/sw.js'
or
[HPM] Error occurred while trying to proxy request / from magento-venia.local.pwadev:8000 to https://magento-venia.local.pwadev:8000/ (ENOTFOUND) (https://nodejs.org/api/errors.html#errors_common_system_errors)

any ideas what might be happening here?

**Comment by zetlen** [comment]
@ericerway I'll keep this issue open for tracking, but unless Mike wants to move to GitHub to discuss, I'll just paste chat logs in here.

**Comment by circlesix** [comment]
the content of my `.env` file is as follows:

```
// file: venia-pwa-concept/theme-frontend-venia/.env

MAGENTO_BACKEND_DOMAIN="http://magento222.test/"
MAGENTO_BACKEND_PUBLIC_PATH="/pub/static/frontend/Magento/venia/en_US"
SERVICE_WORKER_FILE_NAME="sw.js"
```
I modified the `webpack.config.js` to have the id read as `id: 'magento222'`:
```
config.devServer = await PWADevServer.configure({
        publicPath: process.env.MAGENTO_BACKEND_PUBLIC_PATH,
        backendDomain: process.env.MAGENTO_BACKEND_DOMAIN,
        serviceWorkerFileName: process.env.SERVICE_WORKER_FILE_NAME,
        paths: themePaths,
        id: 'magento222'
});
```
This gives me the output when i run `npm start` of:

```
> theme-frontend-venia@0.1.0 start /Applications/MAMP/htdocs/venia-pwa-concept/theme-frontend-venia
> webpack-dev-server --progress --color --env.phase development

Emitting no ServiceWorker in development phase. To enable development phase for ServiceWorkers, pass `enableServiceWorkerDebugging: true` to the ServiceWorkerPlugin configuration.
 10% building modules 1/1 modules 0 active[HPM] Proxy created: [ '**', '!**/*.{js,map,css,json,svg}' ]  ->  http://magento222.test/
[HPM] Subscribed to http-proxy events:  [ 'proxyRes', 'error', 'close' ]

Project is running at https://magento222.local.pwadev:8001/
webpack output is served from https://magento222.local.pwadev:8001/pub/static/frontend/Magento/venia/en_US
Hash: d7b86c0c4ea5098d6880                                                              
Version: webpack 3.11.0
Time: 4800ms
                               Asset       Size    Chunks                    Chunk Names
77fd599ce936468c7fea936dabacccde.svg     414 kB            [emitted]  [big]  
         CMS-73619b0cbfd0cfc58218.js     3.3 kB       CMS  [emitted]         CMS
    Category-2084432fb7e843420284.js     549 kB  Category  [emitted]  [big]  Category
     Product-687fc4347aa00ad4491a.js     701 kB   Product  [emitted]  [big]  Product
                           client.js    3.75 MB    client  [emitted]  [big]  client
                 roots-manifest.json  925 bytes            [emitted]         
[./node_modules/@magento/peregrine/dist/index.js] ./node_modules/@magento/peregrine/dist/index.js 146 bytes {client} [built]
[./node_modules/loglevel/lib/loglevel.js] ./node_modules/loglevel/lib/loglevel.js 7.86 kB {client} [built]
[./node_modules/url/url.js] ./node_modules/url/url.js 23.3 kB {client} [built]
[./node_modules/webpack-dev-server/client/index.js?https://magento222.local.pwadev:8001] (webpack)-dev-server/client?https://magento222.local.pwadev:8001 8.54 kB {client} [built]
[./node_modules/webpack-dev-server/client/overlay.js] (webpack)-dev-server/client/overlay.js 3.67 kB {client} [built]
[./node_modules/webpack/hot/dev-server.js] (webpack)/hot/dev-server.js 2.22 kB {client} [built]
[./node_modules/webpack/hot/emitter.js] (webpack)/hot/emitter.js 77 bytes {client} [built]
   [0] multi (webpack)-dev-server/client?https://magento222.local.pwadev:8001 webpack/hot/dev-server ./src/index.js 52 bytes {client} [built]
[./node_modules/webpack/hot/log-apply-result.js] (webpack)/hot/log-apply-result.js 1.31 kB {client} [built]
[./node_modules/webpack/hot/log.js] (webpack)/hot/log.js 1.04 kB {client} [built]
[./src/RootComponents/CMS/index.js] ./src/RootComponents/CMS/index.js 117 bytes {CMS} [built]
[./src/RootComponents/Category/index.js] ./src/RootComponents/Category/index.js 126 bytes {Category} [built]
[./src/RootComponents/Product/index.js] ./src/RootComponents/Product/index.js 124 bytes {Product} [built]
[./src/index.css] ./src/index.css 1.04 kB {client} [built]
[./src/index.js] ./src/index.js 1.24 kB {client} [built]
    + 328 hidden modules

WARNING in asset size limit: The following asset(s) exceed the recommended size limit (250 kB).
This can impact web performance.
Assets: 
  77fd599ce936468c7fea936dabacccde.svg (414 kB)
  Category-2084432fb7e843420284.js (549 kB)
  Product-687fc4347aa00ad4491a.js (701 kB)
  client.js (3.75 MB)

WARNING in entrypoint size limit: The following entrypoint(s) combined asset size exceeds the recommended limit (250 kB). This can impact web performance.
Entrypoints:
  client (3.75 MB)
      client.js

webpack: Compiled with warnings.
```
When i hit the url  `http://magento222.test/` the app is not hit, when i hit the url `https://magento222.local.pwadev:8001/` i get this output in the console: 

```
[HPM] GET / -> http://magento222.test/
[HPM] Error occurred while trying to proxy request / from magento222.local.pwadev:8001 to http://magento222.test/ (ECONNREFUSED) (https://nodejs.org/api/errors.html#errors_common_system_errors)
```

**Comment by shakyShane** [comment]
@circlesix is your Magento store running correcty at `http://magento222.test/`? 

Can you paste the output of running the following command?

```bash
curl -I http://magento222.test/
```

**Comment by circlesix** [comment]
@shakyShane
It does, i have a custom set up with MAMP so that i can run a dev site on a custom local url. This is necessary in magento 2 as running API tests with swagger wont run on MAMP's default local url structure. 

To your question, here is that output:
```
HTTP/1.1 200 OK
Date: Mon, 07 May 2018 21:05:17 GMT
Server: Apache/2.2.31 (Unix) mod_wsgi/3.5 Python/2.7.12 PHP/7.0.13 mod_ssl/2.2.31 OpenSSL/1.0.2j DAV/2 mod_fastcgi/2.4.6 mod_perl/2.0.9 Perl/v5.24.0
X-Powered-By: PHP/7.0.13
Set-Cookie: store=default; expires=Tue, 07-May-2019 21:05:19 GMT; Max-Age=31536000; path=/; HttpOnly
Set-Cookie: PHPSESSID=08f0f13361dbe93267b540b0c55fd76a; expires=Mon, 07-May-2018 22:05:19 GMT; Max-Age=3600; path=/; domain=magento222.test; HttpOnly
Expires: Sun, 07 May 2017 21:05:19 GMT
Cache-Control: max-age=0, must-revalidate, no-cache, no-store
Pragma: no-cache
X-Content-Type-Options: nosniff
X-XSS-Protection: 1; mode=block
X-Frame-Options: SAMEORIGIN
X-UA-Compatible: IE=edge
Content-Type: text/html; charset=UTF-8
```
BUT! I might have figured out where the url is coming from, I was adding in a new dev site which requires me to change my `/etc/hosts` file. Inside that file, there is this written in:

```
## vagrant-hostmanager-end

127.0.0.1 magento-venia.local.pwadev
127.0.0.1 magento222.local.pwadev
```

So this is where that's coming from. I haven't tested to see if this resolves the issue yet, but I will look into it and get back as soon as i can.

**Comment by circlesix** [comment]
I have removed the referenced to the site domain in my `/etc/hots` file, and when i run `npm` start i get this:
```
> theme-frontend-venia@0.1.0 start /Applications/MAMP/htdocs/venia-pwa-concept/theme-frontend-venia
> webpack-dev-server --progress --color --env.phase development

Emitting no ServiceWorker in development phase. To enable development phase for ServiceWorkers, pass `enableServiceWorkerDebugging: true` to the ServiceWorkerPlugin configuration.
 10% building modules 1/1 modules 0 active[HPM] Proxy created: [ '**', '!**/*.{js,map,css,json,svg}' ]  ->  http://magento222.test
[HPM] Subscribed to http-proxy events:  [ 'proxyRes', 'error', 'close' ]
events.js:167
      throw er; // Unhandled 'error' event
      ^

Error: getaddrinfo ENOTFOUND magento-venia.local.pwadev
    at GetAddrInfoReqWrap.onlookup [as oncomplete] (dns.js:50:26)
Emitted 'error' event at:
    at Server.emit (/Applications/MAMP/htdocs/venia-pwa-concept/theme-frontend-venia/node_modules/spdy/lib/spdy/server.js:237:40)
    at GetAddrInfoReqWrap.doListen [as callback] (net.js:1488:12)
    at GetAddrInfoReqWrap.onlookup [as oncomplete] (dns.js:50:17)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! theme-frontend-venia@0.1.0 start: `webpack-dev-server --progress --color --env.phase development`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the theme-frontend-venia@0.1.0 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/circlesix/.npm/_logs/2018-05-08T00_55_10_063Z-debug.log
```
I have search every where high and low for `magento-venia.local.pwadev` and it's just not there. So i'm back to trying to find out where this value is set in the app. I think it's from the configs for Vagrent, but i would assume that you could use this theme without Vargrent.

**Comment by DrewML** [comment]
@circlesix is this resolved?

**Comment by circlesix** [comment]
No, i tried to go about this in a couple of different ways to try and get my head around what might be going on. But removing this file doesn't correct the url. It still wants to hit `https://magento-venia.local.pwadev:8000/` when it recreates the file. I set this up using the recommended Vagrant box, and i'm getting the same results, with the same url issues, so i don't think this is a issue with MAMP. 

I'm at a loss, but i think i'm going to keep playing with it till i track down some kind of rhyme or reason.

**Comment by DrewML** [comment]
@circlesix let me know if you want to hop on a call and debug. Early adopters give us the benefit of finding our bugs early, so we want to help.

**Comment by circlesix** [comment]
@DrewML, that would be great! If you are on Austin time, i'm in Denver so i think i'm an hour behind you. Tomorrow morning is busy, but i'm can make time work for you, can't pass a chance to trying and move this project along!

**Comment by DrewML** [comment]
I have the afternoon open after 2pm CDT tomorrow. Shoot me a message on the community engineering Slack and we'll figure out a good time.

**Comment by circlesix** [comment]
With @DrewML help i was able to track down the issue with the install, I was using the 2.2.4-dev version of magento that doesn't have the correct ext to run the pwd app. I tracked down the reason for this in the project instructions. 

In step 2 of the instructions you change the install version of magento for the vagrant box project to `ce: "git@github.com:magento/magento2.git::2.3-develop"`. But when you run the install, you end up with the 2.2.4-dev version and not 2.3-dev. I tried changing this to `2.3-dev` (which is the name of the repo) and just `2.3` but that wrecks the install to a non-specified version on a odd branch. 

I ended up doing the normal install of the vagrant box and then manually upgraded magento to the 2.3-dev branch and installed the ext with composer. After that the right modules were in place and the app is running. 

I think this issue is good to close.
